numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -220,6 +220,7 @@ array_implement_array_function_internal(
 {
     PyObject *implementing_args[NPY_MAXARGS];
     PyObject *array_function_methods[NPY_MAXARGS];
+    PyObject *types = NULL;
```

Comment:
In this case, you're forced to put this at the top because `goto cleanup` could have skipped the initialization below.

Reply:
Exactly, that's why I did it like that now, I missed it the first time though.